addons/github/models.py

```diff
@@ -185,6 +185,29 @@ def is_private(self):
         except GitHubError:
             return
 
+    def get_folders(self, **kwargs):
+        if self.has_auth:
+            connection = GitHubClient(external_account=self.external_account)
+            # Since /user/repos excludes organization repos to which the
+            # current user has push access, we have to make extra requests to
+            # find them
+            try:
+                repos = itertools.chain.from_iterable((connection.repos(), connection.my_org_repos()))
+                repo_data = [
+                    {
+                        'addon': 'github',
+                        'kind': 'repo',
+                        'id': repo.id,
+                        'name': repo.name,
+                        'path': os.path.join(repo.owner.login, repo.name)
+                    }
+                    for repo in repos]
+            except GitHubError:
+                repo_data = []
```

Comment:
Do you know if this is consistent with what we do for other addons? Returning an empty list instead of passing up the error to the front-end seems a bit off.

Reply:
For awareness, returning an `HTTPError` here would result in the addons page failing to resolve since `get_folders` is called in `to_json` which is called by the [`/project/<pid>/addons/`](https://github.com/CenterForOpenScience/osf.io/blob/develop/website/project/views/node.py#L331) view:
![screen shot 2018-05-04 at 12 07 02 pm](https://user-images.githubusercontent.com/7913604/39649263-39f02e1a-4fb3-11e8-838a-0e46c67d13c1.png)
 

Good to leave as-is for now and we'll clean up in the linked ticket or when v1 is no longer in use.